viewing_party/party.py

```diff
@@ -1,23 +1,184 @@
+            rating_list.append(watch_list[i]["rating"])
+            # a list contains elements, each elements as a key-value pair
+            # watch_list[i]["rating"] only access the value from the key names "rating"
+    for rate in rating_list:
+        # access every rate inside the rating list
+        ratings_sum += rate
+        # add the rates together
+    average = ratings_sum/len(rating_list)
+    return average
+    
+    
+def get_most_watched_genre(user_data):
+    popular_genre = []
+    genre_list = user_data["watched"]
+    # print(genre_list)
+    if len(genre_list) == 0:
+        return None
+    else:
+       for j in range (len(genre_list)):
+            popular_genre.append(genre_list[j]["genre"])
+            # print(popular_genre)
+    return max(set(popular_genre), key = popular_genre.count)
+
+
 
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
+def get_unique_watched(user_data):
+    # tried to use set and .difference() method to solve,
+    # but finally found out need to return a list
+    friends_list = []
+    unique_list = []
+    friend_watched = user_data["friends"]
+    user_watched = user_data["watched"]
+
+    for val in friend_watched:
+        # read all the "watched" in friends watched list
+        for movie in val["watched"]:
+            # read every title/genre/rating in each watched
+            friends_list.append(movie["title"])
+            # put the name of the movie into the empty list
+    for title in user_watched:
+        # read every movie in user watched list
+        if title["title"] not in friends_list: 
+            unique_list.append(title)
+    return unique_list   
+
+
+def get_friends_unique_watched(user_data):
+    user_watched_list = []
+    friends_unique_movies = []
+    friend_watched = user_data["friends"]
+    user_watched = user_data["watched"]
+
+    for title in user_watched:
+        user_watched_list.append(title["title"])
+    for watched in friend_watched: 
+        for title in watched["watched"]:
+            if title["title"] not in user_watched_list and\
+               title not in friends_unique_movies:
+               # and statement make sure the friends_unique_movies list
+               # won't have duplicated elements 
+                friends_unique_movies.append(title)
+    return friends_unique_movies   
+
+
+
+
         
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
+def get_available_recs(user_data):
+    """ use helper function from get_friends_unique_watched """
+
+    recommended_movie = []
+    subscriptions_list = user_data["subscriptions"]
+    friens_unique_watched = get_friends_unique_watched(user_data)
+    # call the funtion 'get_friends_unique_watched' 
+    # and assign it to a new variable
+
+    for i in range(len(friens_unique_watched)):

```

Comment:
Instead of using `for i in range`, it would be better to use `for movie in friens_unique_watched`, like what you did for the functions before this.